src/chatClient.cpp

```diff
@@ -2376,9 +2461,16 @@ GroupChatRoom::Member::Member(GroupChatRoom& aRoom, const uint64_t& user, chatd:
         if (buf && !buf->empty())
         {
             self->mEmail.assign(buf->buf(), buf->dataSize());
-            if (!self->mRoom.isInitializing() && !self->mRoom.hasTitle() && self->mName.size() <= 1)
+            if (self->mName.size() <= 1)
```

Comment:
Careful, here it's the handler for the attribute *email*, not related to the name. Do we want to trigger the `makeTitleFromMemberNames()` when we retrieve the emails, or only for names?

Reply:
We only execute this code if `self->mName.size() <= 1`. I understand that this mean that we don't have name, then the group name it is made with email

Reply:
Are you sure the mName.size() <= 1 check is safe while the chatroom is initializing (i.e. the code is called from the GroupChatRoom ctor)?

Reply:
@alxvasilev why do you think the `mName` variable could be not initialized yet? That var is part of the initialization list of the `GroupChatRoom::Member` ctor. Furthermore, the check aforementioned is in the callback for a _user attribute fetch_, so it's not called inside the ctor.
Please, check it again and let us know your reason

Reply:
mName is initialized to mName("\0", 1) in the initialization list, i.e. "empty". The code is in the attrib fetch callback, but it may get executed synchronously inside the constructor if the attribute is readily available in memory. That isInitializing() check is there for a reason.

Reply:
:thinking: I see what you mean. In the case where the name is a fail in cache and requires an API request, but the email is a hit and it's solved synchronously inside the ctor, the name would be empty because we are still fetching it.
Therefore, the best would be to serialize the attribute fetches. First, fetch the name. And later, in the callback for the name, fetch the email. That way, we ensure the name is available before checking it in the callback for the email.
@alber2510 please, do the amendments as above.
Thanks, Alex